src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -90,7 +97,7 @@ interface ItemListLayoutUserSettings {
 @observer
 export class ItemListLayout extends React.Component<ItemListLayoutProps> {
   static defaultProps = defaultProps as object;
-
+  @observable columnProps = new Set<string>();
```

Comment:
I think a better name is in order here, since you aren't actually using this data as `props` anywhere.
```suggestion
  @observable hiddenColumnNames = new Set<string>();
```